### ISSUE TITLE
Avoid dangling files on command graphs

### ISSUE THREAD

**Issue by eunovm** [17451k/clade#65]
When I executed commands like in #63, I got the following:
[cmd_graph.dot.pdf](https://github.com/17451k/clade/files/2896649/cmd_graph.dot.pdf)

There is dangling file _/tmp/ccYgfxYM.s_ that likely arises from:
```
/home/novikov/work/klever-data/linux-3.14.79||1012||/usr/bin/gcc||gcc||-x||c||-||-c||-o||/dev/null
/home/novikov/work/klever-data/linux-3.14.79||1014||/usr/lib64/gcc/x86_64-suse-linux/4.8/cc1||/usr/lib64/gcc/x86_64-suse-linux/4.8/cc1||-quiet||-||-quiet||-dumpbase||-||-mtune=generic||-march=x86-64||-auxbase-strip||/dev/null||-o||/tmp/ccYgfxYM.s
/home/novikov/work/klever-data/linux-3.14.79||1014||/usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../x86_64-suse-linux/bin/as||/usr/lib64/gcc/x86_64-suse-linux/4.8/../../../../x86_64-suse-linux/bin/as||--64||-o||/dev/null||/tmp/ccYgfxYM.s
```

I guess that corresponding _cc1_ or/and _as_ build commands should be filtered out by using pids and specified input/output filters.

**Comment by eunovm** [comment]
I do not know if it is a small improvement or a considerable bug.

**Comment by 17451k** [comment]
The issue #63 is a cause of this. When it will be fixed these dangling files will go away.

**Comment by 17451k** [comment]
This is more difficult issue then I thought initially. cc1 command can't be filtered using pid graph because its parent command - gcc -x c - -c -o /dev/null - where already filtered by "Common.filter_out" option. 

"Common.filter_out" option removes commands from Clade database entirely, so they can't be used in filtering by pid graph.

**Comment by 17451k** [comment]
This particular manifestation of the issue (dangling file /tmp/ccYgfxYM.s) is fixed in b2b76d655fec4ea3be1b43b948c461bdc0cc85b9, but the true issue remains unfixed.

**Comment by eunovm** [comment]
> This is more difficult issue then I thought initially. cc1 command can't be filtered using pid graph because its parent command - gcc -x c - -c -o /dev/null - where already filtered by "Common.filter_out" option.
> 
> "Common.filter_out" option removes commands from Clade database entirely, so they can't be used in filtering by pid graph.

So, it seems that Clade needs another approach for filtering. Perhaps filters should play selectively. For instance, Clade should apply them when it outputs nice pictures for users. Besides, they can help to reduce useless source queries as well as useless content in call graphs that can take much time to obtain and to use as well some extra disk space and memory.

But you need to keep all intercepted build commands which are of interest. I guess that this will not take many resources.
